#312 validate unkown query params

linhvo opened the pull request.

yuce commented on handler.go:
```diff
@@ -843,6 +844,12 @@ func (h *Handler) readProtobufQueryRequest(r *http.Request) (*QueryRequest, erro
 // readURLQueryRequest parses query parameters from URL parameters from r.
 func (h *Handler) readURLQueryRequest(r *http.Request) (*QueryRequest, error) {
 	q := r.URL.Query()
+	validQuery := h.getValidURLQuery(r)
```
Since the value of `validQuery` is always the same, would it make sense to save it in a global variable ?

yuce commented on handler.go:
```diff
@@ -875,6 +882,20 @@ func (h *Handler) readURLQueryRequest(r *http.Request) (*QueryRequest, error) {
 	}, nil
 }
 
+func (h *Handler) getValidURLQuery(r *http.Request) map[string]bool {
```
Since this function doesn't depend on a Handler, would it make sense to make it a standalone function? Also name of the function is a bit misleading. Would something like `validURLQueryArgs` work better ?

yuce commented on handler.go:
```diff
@@ -875,6 +882,20 @@ func (h *Handler) readURLQueryRequest(r *http.Request) (*QueryRequest, error) {
 
+func (h *Handler) getValidURLQuery(r *http.Request) map[string]bool {
+	validQuery := make(map[string]bool)
+	args := reflect.ValueOf(QueryRequest{})
```
Small nitpick: since `args.Type()` is used in a loop below, I think it would be better to save it in a var like `argsType` instead of `args`.

linhvo commented:
@yuce: I fixed your comments and have change it a bit. Ss you mentioned to make it a standalone function, I think probably writing this as general for validate any struct that we might use it to validate other params in the future by accept `interface`, instead of specific `QueryRequest`. So I changed it to `validOptions` for general name. How do you think?

yuce commented:
This looks good!
